### DISCUSSION TITLE
Regarding installing spaceone on Microsoft Azure instance

### DISCUSSION THREAD

**Discussion by varun1369** [cloudforet-io/community#114] [Bug Report]
I am getting this issue when installing cloud foret version 1.10.4 on a public VM on Microsoft Azure. Have replaced the public IP in the screenshot of the minikube.yaml shown below.

![Screenshot_20230205_091715](https://user-images.githubusercontent.com/82792752/216883303-dbd1bd80-7e0b-458a-b362-816d4b5687d9.png)

Getting the below error when trying to access the page on my browser.

![Screenshot_20230205_091604](https://user-images.githubusercontent.com/82792752/216883587-37be9ce6-186d-40ab-911e-cf0c2a62f72a.png)

Please could you resolve this issue.

**Comment by GeunjeLEE** [comment]
Hi @varun1369 !
Thank you for you reporting.

It looks like you have incorrect settings. 
current your config has `spaceone` in `DOMAIN_NAME_REF`, but it should be `config` or `hostname` :)

Could you please edit `DOMAIN_NAME_REF` with `config` and deploy the helm release?

---
** WARNING **
Disable spaceone-initializer in minikube.yaml when updating helm release with edited configuration.

ex) 
```diff
spaceone-initializer:
-  enabled: true
+  enabled: false
  image:
      name: spaceone/spacectl
      version: 1.10.4.2
(omit)
```

**Comment by varun1369** [comment]
getting Imagepullbackoff even after changing to the above settings.
![Screenshot_20230206_092500](https://user-images.githubusercontent.com/82792752/217028108-cb66421f-f352-40cf-9cc4-b699a32f86c5.png)

**Reply by GeunjeLEE** [reply]
Hi!

Could you please share the description of the pod getting the Imagepullbackoff error so we can figure out the cause?
The description can printed using the `kubectl description {pod_name}` command

---

+) We are so sorry, It looks like there is a configuration that needs to be disabled.
the `consul-connect-injector` should be disabled.(It can cause the error that the pod hangs.)
To do this, you need to add the settings below to the configuration consul section in minikube.yaml.

ex)
```diff
consul:
    enabled: true
    server:
        replicas: 1
    ui:
        enabled: false
+   connectInject:
+       enabled: false
```

This configuration will be fixed in a new example (1.11.0).

**Comment by varunmanju** [comment]
Still getting the same issue after applying the above configurations in minikube.yaml file for version 14.2. 
![Screenshot_20230205_091604](https://user-images.githubusercontent.com/47963858/218942076-ba47a8d2-217a-4c4f-8eec-1ea36e8d2775.png)

**Comment by varunmanju** [comment]
should I change locahost in this uri to the public ip of the Azure instance.
grpc://localhost:50051/v1/Domain/get_public_key

**Comment by GeunjeLEE** [comment]
Hi, @varunmanju 

`grpc://localhost:50051/v1/Domain/get_public_key` is for networking in kubernetes networks only.

When a user accesses SpaceONE using a browser, the browser gets the console-api address from `production_json.CONSOLE_API.ENDPOINT` and uses it.

So you need to change the ENDPOINT as below and restart(just kill the pod) console pod

```diff
console:
  enabled: true
  developer: false
  name: console
  replicas: 1
  image:
      name: spaceone/console
      version: 1.11.0
  imagePullPolicy: IfNotPresent

  production_json:
      CONSOLE_API:
-       ENDPOINT: http://localhost:8081
+       ENDPOINT: http://{VM_IP_ADDRESS}:{NODE_PORT_FOR_CONSOLE_API}
      DOMAIN_NAME: spaceone
      DOMAIN_NAME_REF: config
```

**Reply by varunmanju** [reply]
yes i did that.and am still facing this issue

**Reply by varunmanju** [reply]
could we get on a call
